lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,151 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.lcevc.Dil');
+goog.require('shaka.log');
+goog.require('shaka.util.IReleasable');
+
+/**
+ * @summary
+ *  lcevcDil - (MPEG-5 Part 2 LCEVC - Decoder Integration Layer) provides
+ *  all the operations related to the enhancement and rendering
+ *  of LCEVC enabled streams and on to a canvas.
+ * @implements {shaka.util.IReleasable}
+ * @export
+ */
+shaka.lcevc.Dil = class {
+  /**
+   * @param {HTMLVideoElement} media The video element that will be attached to
+   * LCEVC Dil for input.
+   * @param {Element} canvas The canvas element that will be attached
+   * to LCEVC Dil to render the enhanced frames.
+   * @param {shaka.extern.LcevcConfiguration} dilConfig The LCEVC DIL
+   * config object to initialize the LCEVC DIL.
+   */
+  constructor(media, canvas, dilConfig) {
+    /** @private {?LcevcDil.LcevcDIL} */
+    this.dil_ = null;
+
+    /** @private {number} */
+    this.variantId_ = -1;
+
+    /** @private {HTMLVideoElement} */
+    this.media_ = media;
+
+    /** @private {Element} */
+    this.canvas_ = canvas;
+
+    /** @private {shaka.extern.LcevcConfiguration} */
+    this.dilConfig_ = dilConfig;
+
+    this.create_();
+  }
+
+  /**
+   * Append data to the LCEVC Dil.
+   * @param {BufferSource} data
+   */
+  appendBuffer(data) {
+    if (this.dil_) {
+      this.dil_.appendBuffer(data, 'video', this.variantId_);
+    }
+  }
+
+  /**
+   * Hide the canvas specifically in the case of a DRM Content
+   */
+  hideCanvas() {
+    if (this.dil_) {
+      this.canvas_.classList.add('shaka-hidden');
+    }
+  }
+
+  /**
+   * Create LCEVC Dil.
+   * @private
+   */
+  create_() {
+    if (this.isSupported_() && !this.dil_) {
```

Comment:
Please add a check for canvas either here or at the constructor call site in Player's createLcevcDil_, since a non-UI app has to attach a canvas explicitly and may not have done so.

Reply:
Added a canvas check under `isSupported()`